MIP/mip-58/README.md

```diff
@@ -0,0 +1,201 @@
+   - **No Funds Requirement**: User is not required to have funds on receiving chain and we do not have to build sponsored transactions.
+   - **Delivery by Relayer**: Because the relayer delivers the funds, there is no requirement for the user to complete the transaction on receiving chain, therefore massively simplifying the user experience and allowing a smooth onboarding on the network for bridges from L1 to L2. User only has to await for the finalization on the source chain and for the relayer to perform the completion on receiving chain. This is standard practice for every major bridge in the ecosystem.
+   - **Less parameters**: Because there is no exchange of secrets between the user and relayer, we have a substantial reduction of logic.
+   - **Only Completable**: Currently we reserve a refunder role to revert transactions. This approach is different where we guarantee delivery of funds through the same party that would guarantee funds being refunded, because bridges can ONLY be completed.
+
+2. **Consolidation of Logic**:
+   - Merge lock and completion functionality on the counterparty contract. Once lock is called, funds are already in the control of the user. On the HTLC implementation, once the timelock is over and complete on initiator has not been called, both the initiator and counterparty funds are available to the user, opening up for an exploit.
+   - Remove refund functionality entirely to eliminate associated exploits.
+
+3. **Parameter Validation**:
+   - Ensure parameter validation on the counterparty to prevent invalid transactions.
+
+4. **Relayer Redundancy**:
+   - Use two types of relayers:
+     - **Automated Relayer**: Operated with minimal human involvement; its private key is highly secured.
+     - **Multi-Signature Relayer**: Managed by the team to guarantee transaction completion in case of failures.
+
+5. **Cost Efficiency**:
+   - Minimize gas costs by reducing the number of interactions and simplifying fee calculations.
+
+6. **Enhanced Security**:
+   - Avoid refund logic to close exploit windows.
+   - Protect against key compromise through key isolation or known by no parties and multi-signature relayer setups.
+   - There is no scenario where a bridge could lead to double-spending. It's either completed by relayers or not.
+   - User currently can loose its bridge `preImage` which could lead to them being unable to complete the bridge. By not relying on a `preImage` from the user, it minimizes issues. It is not a loss in security because the purpose of the `preImage` is solely for refunding.
+
+7. **Batch Processing for Downtime**:
+   - Multi-signature relayers can process multiple pending transactions in a single batch during downtime.
+
+8. **BridgeTransferId**:
+   - Continue using unique identifier to prevent double-spending and track transactions securely.
+
+9. **Bridge Fee**:
```

Comment:
https://github.com/movementlabsxyz/movement/issues/781

Reply:
@apenzk @franck44 wondering if you feel like this is a valid approach.

Reply:
responded at the link you shared